src/Microsoft.Diagnostics.Monitoring/DiagnosticServices.cs

```diff
@@ -39,71 +47,88 @@ public IEnumerable<int> GetProcesses()
             }
 
-            if (!pid.HasValue)
+            int pidValue = GetSingleProcessId(pid);
+
+            DiagnosticsMonitor monitor = new DiagnosticsMonitor(_serviceProvider, new CpuProfileConfiguration());
+            Stream stream = await monitor.ProcessEvents(pidValue, durationSeconds, cancellationToken);
+
+            return CreateResult<IStreamResult>(pidValue, new StreamResult(monitor, stream));
+        }
+
+        public async Task<OperationResult<IStreamResult>> StartTrace(int? pid, int durationSeconds, CancellationToken token)
+        {
+            if ((durationSeconds < 1) || (durationSeconds > MaxTraceSeconds))
             {
-                pid = GetSingleProcessId();
+                throw new InvalidOperationException("Invalid duration");
             }
 
-            //TODO Should we limit only 1 trace per file?
-            var client = new DiagnosticsClient(pid.Value);
+            int pidValue = GetSingleProcessId(pid);
 
-            //TODO Pull event providers from the configuration.
-            var cpuProviders = new EventPipeProvider[] {
-                new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler", System.Diagnostics.Tracing.EventLevel.Informational),
-                new EventPipeProvider("Microsoft-Windows-DotNETRuntime", System.Diagnostics.Tracing.EventLevel.Informational, (long)Tracing.Parsers.ClrTraceEventParser.Keywords.Default)
-            };
+            DiagnosticsMonitor monitor = new DiagnosticsMonitor(_serviceProvider, new LoggingSourceConfiguration());
+            Stream stream = await monitor.ProcessEvents(pidValue, durationSeconds, token);
+            return CreateResult<IStreamResult>(pidValue, new StreamResult(monitor, stream));
+        }
+
+        public async Task<OperationResult> StartLogs(Stream outputStream, int? pid, int durationSeconds, CancellationToken token)
+        {
+            if ((durationSeconds < 1) || (durationSeconds > MaxTraceSeconds))
+            {
+                throw new InvalidOperationException("Invalid duration");
+            }
+
+            int pidValue = GetSingleProcessId(pid);
+
+            //We want to use ILogger since at some point in the future we may want to push this data to other
+            //sinks. At the same time we want these to be transient for the purposes of streaming the data, but not for other logs (such as those of
+            //the rest service itself).
+            ServiceCollection serviceCollection = new ServiceCollection();
+            serviceCollection.AddLogging(loggingBuilder => loggingBuilder.AddProvider(new StreamingLoggerProvider(outputStream)));
 
-            EventPipeSession session = client.StartEventPipeSession(cpuProviders, requestRundown: true);
+            //TODO This is not ideal since the newly added services do not actually know about the parent service provider.
+            var serviceProvider = new AggregateServiceProvider(_serviceProvider, serviceCollection.BuildServiceProvider());
```

Comment:
I don't really like this but I'm not sure of a better alternative at this point. Adding ILogger at a broader scope mixes the ILogger used for logging vs the one used for capturing data.